apollo-federation/src/sources/connect/header.rs

```diff
@@ -145,28 +149,42 @@ impl From<VariableParseError<Span<'_>>> for HeaderValueError {
                 namespace,
                 location,
             },
+            VariableParseError::InvalidHeaderValue { value, location } => {
+                HeaderValueError::ParseError {
+                    message: format!("invalid HTTP header value `{value}`"),
+                    location,
+                }
+            }
         }
     }
 }
 #[derive(Debug, PartialEq, Clone)]
 enum HeaderValuePart<'a> {
-    Text(String),
+    Constant(http::HeaderValue),
     Variable(VariableReference<'a, Namespace>),
 }
 
 impl<'a> HeaderValuePart<'a> {
     fn parse(input: Span<'a>) -> IResult<Span<'a>, Self, VariableParseError<Span>> {
         alt((
             map(variable_reference, Self::Variable),
-            map(map(text, |s| s.fragment().to_string()), Self::Text),
+            map(parse_header_value, Self::Constant),
         ))(input)
     }
 }
 
 type Span<'a> = LocatedSpan<&'a str>;
 
-fn text(input: Span) -> IResult<Span, Span, VariableParseError<Span>> {
-    recognize(many1(none_of("{}")))(input)
+fn parse_header_value(input: Span) -> IResult<Span, http::HeaderValue, VariableParseError<Span>> {
```

Comment:
This is where you should use a new `HeaderParseError` type, or something like that. One variant of that might be `VariableError`, and another might be `ValueError` or something. This was just returning `VariableParseError` because variable errors were the only thing that could go wrong previously.